apps/editing-toolkit/editing-toolkit-plugin/block-patterns/class-block-patterns.php

```diff
@@ -45,112 +62,116 @@ public static function get_instance() {
-		// Register Block Pattern Categories.
+	private function register_patterns() {
+		// Remove core pattern categories, categories added later that match
+		// will then remain in alphabetical order.
 		if ( class_exists( 'WP_Block_Pattern_Categories_Registry' ) ) {
-			register_block_pattern_category( 'blog', array( 'label' => _x( 'Blog', 'Block pattern category', 'full-site-editing' ) ) );
-			register_block_pattern_category( 'call-to-action', array( 'label' => _x( 'Call to Action', 'Block pattern category', 'full-site-editing' ) ) );
-			register_block_pattern_category( 'contact', array( 'label' => _x( 'Contact', 'Block pattern category', 'full-site-editing' ) ) );
-			register_block_pattern_category( 'images', array( 'label' => _x( 'Images', 'Block pattern category', 'full-site-editing' ) ) );
-			register_block_pattern_category( 'list', array( 'label' => _x( 'List', 'Block pattern category', 'full-site-editing' ) ) );
-
-			// The 'Two Columns of Text' pattern is in the 'columns' and 'text' categories.
-			// Removing 'columns' so it doesn't appear as a category with only a single item.
+			if ( \WP_Block_Pattern_Categories_Registry::get_instance()->is_registered( 'buttons' ) ) {
+				unregister_block_pattern_category( 'buttons' );
+			}
 			if ( \WP_Block_Pattern_Categories_Registry::get_instance()->is_registered( 'columns' ) ) {
 				unregister_block_pattern_category( 'columns' );
 			}
+			if ( \WP_Block_Pattern_Categories_Registry::get_instance()->is_registered( 'gallery' ) ) {
+				unregister_block_pattern_category( 'gallery' );
+			}
+			if ( \WP_Block_Pattern_Categories_Registry::get_instance()->is_registered( 'header' ) ) {
+				unregister_block_pattern_category( 'header' );
+			}
+			if ( \WP_Block_Pattern_Categories_Registry::get_instance()->is_registered( 'text' ) ) {
+				unregister_block_pattern_category( 'text' );
+			}
 		}
 
 		if ( class_exists( 'WP_Block_Patterns_Registry' ) ) {
-			// Remove core patterns except 'Two Columns of Text'.
-			// Unfortunately, \WP_Block_Patterns_Registry::get_instance()->get_all_registered() doesn't return the pattern names as keys.
+			// Remove core patterns.
 			foreach ( \WP_Block_Patterns_Registry::get_instance()->get_all_registered() as $pattern ) {
-				if ( 'core/' === substr( $pattern['name'], 0, 5 ) && 'core/text-two-columns' !== $pattern['name'] ) {
+				if ( 'core/' === substr( $pattern['name'], 0, 5 ) ) {
 					unregister_block_pattern( $pattern['name'] );
 				}
 			}
+		}
 
-			// Add our patterns.
-			foreach ( $this->get_patterns() as $name => $pattern ) {
-				register_block_pattern( $name, $pattern );
+		$pattern_categories = array();
+		$block_patterns     = $this->get_patterns();
+
+		foreach ( (array) $this->get_patterns() as $pattern ) {
+			foreach ( (array) $pattern['categories'] as $slug => $category ) {
+				$pattern_categories[ $slug ] = $category['title'];
 			}
 		}
+
+		// Order categories alphabetically and register them.
+		ksort( $pattern_categories );
```

Comment:
Does it sort by slug or by label? Slug sorting would work in English but not in other languages.

Reply:
Good catch, switching to `asort()`. :+1: